profiling/src/internal/profile/mod.rs

```diff
@@ -137,6 +192,28 @@ impl Profile {
         Ok(())
     }
 
+    pub fn resolve(&mut self, id: ManagedStringId) -> anyhow::Result<StringId> {
+        let non_empty_string_id = if let Some(valid_id) = NonZeroU32::new(id.value) {
+            valid_id
+        } else {
+            return Ok(StringId::ZERO); // Both string tables use zero for the empty string
+        };
+
+        let string_id = self.string_storage
+            .as_ref()
+            // Safety: We always get here through a direct or indirect call to add_string_id_sample,
+            // which already ensured that the string storage exists.
+            .ok_or_else(|| anyhow::anyhow!("Current sample makes use of ManagedStringIds but profile was not created using a string table"))?
+            .read()
+            // Safety: This failure is unlikely as it only happens if the lock is poisoned (and for
+            // the lock to become poisoned, another unlikely failure already happened
+            // before)
+            .expect("acquisition of read lock on string storage should succeed")
```

Comment:
Still might as well return an error rather than kill the process

Reply:
Fixed in 432c12e0455e2f3784dd63aec7ed4aba26742fba